### PR TITLE
Add support for searching trainers in the Showcase project

### PR DESCRIPTION
**Pull request by maxime-poulain**
This PR is a follow up of #124 
It refactors the code of the Trainer list in a service to separate concerns.

### CODE REVIEW

**Comment by maxime-poulain** [comment]
@Victor-van-Duijnen

**Comment by maxime-poulain** [comment]
@LievenPraet 
@Victor-van-Duijnen 
This little PR is blocking me to open another :x; it's a very small one.

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Infrastructure/Data/Extensions/TrainerQueryableExtensions.cs [hunk 1]
```diff
@@ -9,4 +10,15 @@ public static async Task<int> TrainerCountAsync(this IQueryable<TrainerDetails>
...
     }
+
+    public static async Task<PagedList<TrainerList>> SearchPaginatedTrainersAsync(this IQueryable<TrainerList> query, string? searchKeyWord, int pageNumber, int pageSize)
+    {
+        if (!string.IsNullOrEmpty(searchKeyWord))
+        {
+            query = query.Where(trainerList => trainerList.FirstName.Contains(searchKeyWord) || trainerList.LastName.Contains(searchKeyWord));
+        }
+        var paginationResult = await query.PaginateAsync(pageNumber, pageSize);
+
```
Comment (Victor-van-Duijnen) [inline src/Showcase/src/Smart.FA.Catalog.Showcase.Infrastructure/Data/Extensions/TrainerQueryableExtensions.cs:21]:
```suggestion
        }
        
        var paginationResult = await query.PaginateAsync(pageNumber, pageSize);
```

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Infrastructure/Data/Scripts/v_TrainerList.sql [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+    ,trainer.PofileImagePath
+	,count(assignment.TrainingId) As TrainingCount
+FROM Cfa.Trainer trainer
+    INNER JOIN Cfa.TrainerAssignment assignment
+    ON assignment.TrainerId = trainer.Id
+    INNER JOIN Cfa.Training training ON
+    assignment.TrainingId = training.Id
+WHERE training.TrainingStatusTypeId = 3
+GROUP BY trainer.Id, trainer.FirstName, trainer.LastName, trainer.Title, trainer.ProfileImagePath
+HAVING count(assignment.TrainingId) > 1
```
Comment (Victor-van-Duijnen) [inline src/Showcase/src/Smart.FA.Catalog.Showcase.Infrastructure/Data/Scripts/v_TrainerList.sql:17]: Should it not select trainers with at least one training? So rather: 
```suggestion
HAVING count(assignment.TrainingId) >= 1
```

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Mappers/TrainerMapper.cs [hunk 3]
```diff
@@ -21,4 +23,16 @@ public static List<TrainerSocialNetwork> ToTrainerSocialNetworks(this List<Domai
         }
         return socialNetworks;
```
Comment (Victor-van-Duijnen) [inline src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Mappers/TrainerMapper.cs:24]:
```suggestion
        }
        
        return socialNetworks;
```

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Trainer/TrainerList/TrainerList.cshtml [hunk 4]
```diff
@@ -16,43 +16,46 @@
...
+        <form method="get">
+            <div class="c-input-with-icon">
+                <input class="c-input c-input--max-width" id="searchKeyword" name="searchKeyword" type="search" placeholder="Search…">
+                <div class="o-svg-icon o-svg-icon-search  ">
+                    <svg width="24" height="24" viewBox="0 0 24 24" fill="#1267FB" xmlns="http://www.w3.org/2000/svg">
+                        <path d="M10 4C6.68629 4 4 6.68629 4 10C4 13.3137 6.68629 16 10 16C13.3137 16 16 13.3137 16 10C16 6.68629 13.3137 4 10 4ZM2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10C18 11.8487 17.3729 13.551 16.3199 14.9056L21.7071 20.2929C22.0976 20.6834 22.0976 21.3166 21.7071 21.7071C21.3166 22.0976 20.6834 22.0976 20.2929 21.7071L14.9056 16.3199C13.551 17.3729 11.8487 18 10 18C5.58172 18 2 14.4183 2 10Z" fill="#1267FB"></path>
+                    </svg>
+                </div>
+            </div>
+        </form></section>
```
Comment (Victor-van-Duijnen) [inline src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Trainer/TrainerList/TrainerList.cshtml:30]:
```suggestion
        </form>
    </section>
```

File: src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Trainer/TrainerList/TrainerList.cshtml [hunk 5]
```diff
@@ -66,6 +69,7 @@
                 </div>
 
                 <div>Page @(Model?.CurrentPage)/@(Model?.Trainers.TotalPages ?? 0)</div>
+
                 <div class="c-toolbar__right">
                     @if (Model?.Trainers.HasNextPage ?? false)
                     {
```
Comment (Victor-van-Duijnen) [inline src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Trainer/TrainerList/TrainerList.cshtml:75]: Can't comment that but there are too many newlines at the end of that file.

**Review by Victor-van-Duijnen — CHANGES_REQUESTED** [review]

**Pull request merged**